txnsync/encodedgroupsunmarshalers.go

```diff
@@ -0,0 +1,644 @@
+			return err
+		}
+		signedTxns[i].Txn.Type = ByteToTxType(b)
+		return nil
+	})
+	for i := range signedTxns {
+		if signedTxns[i].Txn.Type == "" {
+			signedTxns[i].Txn.Type = ByteToTxType(stub.TxTypeOffset)
+		}
+	}
+	if err != nil {
+		return err
+	}
+
+	if err := stub.reconstructTxnHeader(signedTxns, genesisID, genesisHash); err != nil {
+		return fmt.Errorf("failed to reconstructTxnHeader: %w", err)
+	}
+	if err := stub.reconstructKeyregTxnFields(signedTxns); err != nil {
+		return fmt.Errorf("failed to reconstructKeyregTxnFields: %w", err)
+	}
+	if err := stub.reconstructPaymentTxnFields(signedTxns); err != nil {
+		return fmt.Errorf("failed to reconstructPaymentTxnFields: %w", err)
+	}
+	if err := stub.reconstructAssetConfigTxnFields(signedTxns); err != nil {
+		return fmt.Errorf("failed to reconstructAssetConfigTxnFields: %w", err)
+	}
+	if err := stub.reconstructAssetTransferTxnFields(signedTxns); err != nil {
+		return fmt.Errorf("failed to reconstructAssetTransferTxnFields: %w", err)
+	}
+	if err := stub.reconstructAssetFreezeTxnFields(signedTxns); err != nil {
+		return fmt.Errorf("failed to reconstructAssetFreezeTxnFields: %w", err)
+	}
+	if err := stub.reconstructApplicationCallTxnFields(signedTxns); err != nil {
+		return fmt.Errorf("failed to reconstructApplicationCallTxnFields: %w", err)
+	}
+	if err := stub.reconstructCompactCertTxnFields(signedTxns); err != nil {
+		return fmt.Errorf("failed to reconstructCompactCertTxnFields: %w", err)
+	}
+	return nil
+}
+
+func (stub *txGroupsEncodingStub) reconstructTxnHeader(signedTxns []transactions.SignedTxn, genesisID string, genesisHash crypto.Digest) (err error) {
+	err = stub.BitmaskSender.iterate(int(stub.TotalTransactionsCount), len(stub.Sender)/crypto.DigestSize, func(i int, index int) error {
+		return nextSlice(&stub.Sender, signedTxns[i].Txn.Sender[:], crypto.DigestSize)
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskFee.iterate(int(stub.TotalTransactionsCount), len(stub.Fee), func(i int, index int) error {
+		signedTxns[i].Txn.Fee = stub.Fee[index]
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskFirstValid.iterate(int(stub.TotalTransactionsCount), len(stub.FirstValid), func(i int, index int) error {
+		signedTxns[i].Txn.FirstValid = stub.FirstValid[index]
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskLastValid.iterate(int(stub.TotalTransactionsCount), len(stub.LastValid), func(i int, index int) error {
+		signedTxns[i].Txn.LastValid = stub.LastValid[index]
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskNote.iterate(int(stub.TotalTransactionsCount), len(stub.Note), func(i int, index int) error {
+		signedTxns[i].Txn.Note = stub.Note[index]
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskGenesisID.iterate(int(stub.TotalTransactionsCount), int(stub.TotalTransactionsCount), func(i int, index int) error {
+		signedTxns[i].Txn.GenesisID = genesisID
```

Comment:
Why does genesisID have a bitmask?
Can't it just be populate similar to the genesishash?